Implement integrations test for physics plugins

aysunrhn opened the pull request:
Implemented integration tests for the physics plugin system in `src/sedtrails/transport_converter/plugins/physics/`  to ensure new plugins are compliant with the existing structure.

Closes #253 

### Relevant files:
- `tests/transport_converter/plugins/test_physics_plugin.py`:
    - Each plugin inherits from the BasePhysicsPlugin class.
    - Each plugin has an `add_physics` method.
    - Each plugin's `add_physics` method is callable.
    - Each plugin has `sedtrails_data` as parameter. 

- `pyproject.toml`
    - I added a custom marker (called "integration") to mark these tests, so that we can run them separately from the unit tests using `pytest -m integration`.

- `.github/workflows/pytest.yml`
   - Split the pytest action into 2 jobs: unit test and integration test
   - Integration job needs the unit job to run first

aysunrhn committed:
Add integration test for physics plugins

aysunrhn committed:
Add integration test workflow separate from unit tests

aysunrhn committed:
Add test to check add_physics method is callable

aysunrhn committed:
Add integration pytest mark

aysunrhn committed:
Update test name to be consistent

aysunrhn committed:
Add test to check sedtrails_data parameter in plugins

manuGil commented on tests/transport_converter/plugins/test_physics_plugin.py:
```diff
@@ -0,0 +1,81 @@
+    # Should we check the position of the sedtrails_data as well?
+    assert "sedtrails_data" in param_names, (
+        f"{plugin_class.__name__}.add_physics should have 'sedtrails_data' parameter. "
+        f"Found parameters: {param_names}"
```
I think we should check for the order. In the base-plugin it must the the first parameter

```python
    def add_physics(self, sedtrails_data: SedtrailsData, *args, **kwargs):
 ```

manuGil commented on tests/transport_converter/plugins/test_physics_plugin.py:
```diff
@@ -0,0 +1,81 @@
+import pytest
+from sedtrails.transport_converter.plugins.physics.plugin import BasePhysicsPlugin
+
+# TODO: Is there a better way to get the plugin directory?
```
I think the path can be hard-coded because this will only run on the source code. Can you try if tests still work if we hard-code the path to the plugin directory, both when running locally and on GitHub?

manuGil reviewed (approved):
Thanks for this Aysun. please consider my comments before merging.

manuGil commented:
I just noticed that the integration tests failed. Can you fix those please? I also recommend to print the name of the plugin that fails as part of the assertion message. that way it will be easier to identify the 'offender' plugin.

aysunrhn commented:
> I just noticed that the integration tests failed. Can you fix those please? I also recommend to print the name of the plugin that fails as part of the assertion message. that way it will be easier to identify the 'offender' plugin.

Yes it fails because `soulsby.py` does not have the sedtrails_data as input. It is the latest test that I added for the plugins. Should I modify the `soulsby.py` script? It is a placeholder right now anyways.

manuGil commented:
> > I just noticed that the integration tests failed. Can you fix those please? I also recommend to print the name of the plugin that fails as part of the assertion message. that way it will be easier to identify the 'offender' plugin.
> 
> Yes it fails because `soulsby.py` does not have the sedtrails_data as input. It is the latest test that I added for the plugins. Should I modify the `soulsby.py` script? It is a placeholder right now anyways.

I understand now. Natasha's code is still on review. Better not to touch the soulsby.py file for now. Let's keep it as is

aysunrhn committed:
Check for parameter order in physics plugins

aysunrhn committed:
Print plugin file name that does not pass integration tests

aysunrhn commented:
- I modified the plugin parameter test to check that sedtrails_data is in the correct order.
- Updated the print message to state which plugin file is failing the test:
![](https://imgur.com/lpGilTL.png)
